build.sh

```diff
@@ -187,6 +187,14 @@ if buildAll || hasArg libcudf; then
 
     cmake --build . -j${PARALLEL_LEVEL} ${VERBOSE_FLAG}
 
+    # Record build times
+    if [[ -f "${LIB_BUILD_DIR}/.ninja_log" ]]; then
+        echo "Formatting build times"
+        python ${REPODIR}/ci/utils/sort_ninja_log.py ${LIB_BUILD_DIR}/.ninja_log > ${LIB_BUILD_DIR}/ninja_log.csv
```

Comment:
How about this one-liner?
`awk 'BEGIN{ print "name, time";} NR>1 {a=$2-$1; print $4 "," ,a;}' .ninja_log`

If you want to avoid duplicates. (keeps the first one).
`awk 'BEGIN{ print "name, time";} NR>1 && !visited[$4]++ {a=$2-$1; print $4 "," ,a;}' .ninja_log`

Reply:
Sorted, skip duplicates (keep last repeated key).
`tail -n +2 .ninja_log | tac | awk 'BEGIN{ print "time,file";} !visited[$4]++ {a=$2-$1; print a","$4 | "sort -nr" }' > ninja_log.csv`

Reply:
I really don't like this kind of thing because of how difficult it is to figure out what is it doing and how to fix it later on.
The python utility is also useful when trying to track down build times in a local build.